### PR TITLE
Remove `WITH_AWS` option

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [X] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Review by caneGuy — DISMISSED** [review]

**Review by imay — DISMISSED** [review]
LGTM

File: be/test/CMakeLists.txt [hunk 1]
```diff
@@ -271,6 +267,11 @@ if ("${CMAKE_BUILD_TARGET_ARCH}" STREQUAL "x86" OR "${CMAKE_BUILD_TARGET_ARCH}"
         set_source_files_properties(./formats/json/nullable_column_test.cpp PROPERTIES COMPILE_FLAGS -mno-avx2)
 endif()
 
+if (WITH_AWS)
```
Comment (trueeyu) [inline be/test/CMakeLists.txt:270]: remove WITH_AWS?

File: be/test/CMakeLists.txt [hunk 2]
```diff
@@ -117,8 +116,6 @@ set(EXEC_FILES
         io/array_input_stream_test.cpp
         ./io/fd_output_stream_test.cpp
         io/fd_input_stream_test.cpp
-        ./io/s3_output_stream_test.cpp
-        io/s3_input_stream_test.cpp
```
Comment (starrocks-xueli) [inline be/test/CMakeLists.txt:121]: it seems this should not be removed, you can ask zhuming.

File: be/test/CMakeLists.txt (hunk 1 above)
Comment (dirtysalt) [reply be/test/CMakeLists.txt:270]: Some UTs access AWS bucket which can not run in Aliyun Env. So I have to preseve this option to excude this UTs in Aliyun Env.

File: be/test/CMakeLists.txt (hunk 2 above)
Comment (dirtysalt) [reply be/test/CMakeLists.txt:121]: It's preserved in following section.

```
if (WITH_AWS)
    set(EXEC_FILES ${EXEC_FILES} ./io/s3_output_stream_test.cpp ./io/s3_input_stream_test.cpp
        ./env/env_s3_test.cpp)
endif()
```

File: be/test/CMakeLists.txt (hunk 1 above)
Comment (sduzh) [inline be/test/CMakeLists.txt:270]: Please ensure the following tests always be compiled, but skip running them when there is no available S3 environment.

Comment (dirtysalt) [reply be/test/CMakeLists.txt:270]: It's possible but I don't know any plausible solution now. I have to do some search.

Comment (sduzh) [reply be/test/CMakeLists.txt:270]: Run tests with the flag `--gtest-filter=-*S3*` should works.

**Review by trueeyu — DISMISSED** [review]

File: be/test/io/s3_input_stream_test.cpp [hunk 3]
```diff
@@ -40,21 +39,31 @@ class S3InputStreamTest : public testing::Test {
...
+        }
+    }
 
     void TearDown() override {}
 
     std::unique_ptr<S3InputStream> new_random_access_file();
 };
 
 void S3InputStreamTest::SetUpTestCase() {
+    if (!env_util::is_running_on_ec2()) {
```
Comment (sduzh) [inline be/test/io/s3_input_stream_test.cpp:54]: Why check this? S3InputStream/S3OutpuStream can be used to read and write any service compatible with S3 protocol, including OSS.

Comment (sduzh) [reply be/test/io/s3_input_stream_test.cpp:54]: Actually, I'm using OSS as the endpoint when running this test on my local machine.

Comment (sduzh) [reply be/test/io/s3_input_stream_test.cpp:54]: You can keep the `with-aws` flag in run-ut.sh, and if it's false, you can run the unit tests with flag '--gtest-filter=-*S3*'.

Comment (dirtysalt) [reply be/test/io/s3_input_stream_test.cpp:54]: OK. I'll fix for that.

Comment (sduzh) [reply be/test/io/s3_input_stream_test.cpp:54]: Whether the tests are running on EC2 does not matter, what really matters is whether you have a correct S3 configuration in the file `be.conf`. For security reasons, we don't provide an accessible S3 endpoint in `be.conf` by default, that's why the S3 tests are disabled by default. 
But for developers who have configured a valid S3 endpoint and AK/SK, they can run these tests by running the 'run-ut.sh' script with option `--with-aws`.

Comment (dirtysalt) [reply be/test/io/s3_input_stream_test.cpp:54]: Could it be ok to check both ec2 and aliyun ? Or you still think using --gtest_filter=-*S3* is good idea?

Comment (sduzh) [reply be/test/io/s3_input_stream_test.cpp:54]: Sorry, I don't get your point. To run these tests successfully, you need to provide `access_key_id` and `secret_access_key`, how can detecting the running environment help?

Comment (dirtysalt) [reply be/test/io/s3_input_stream_test.cpp:54]: I'll fix by following your idea.

**Comment by dirtysalt** [comment]
run starrocks_fe_unittest

**Review by sduzh — APPROVED** [review]

**Comment by dirtysalt** [comment]
run starrocks_fe_unittest

**Comment by dirtysalt** [comment]
run starrocks_fe_unittest

**Review by stdpain — APPROVED** [review]

**Pull request merged**